akka-http/src/main/scala/akka/http/javadsl/server/directives/RouteDirectives.scala

```diff
@@ -38,14 +37,31 @@ abstract class RouteDirectives extends RespondWithDirectives {
   /**
    * Java-specific call added so you can chain together multiple alternate routes using comma,
    * rather than having to explicitly call route1.orElse(route2).orElse(route3).
+   * @deprecated Use the `RouteDirectives.concat` method instead.
    */
+  @Deprecated
+  @deprecated("Use the RouteDirectives.concat method instead.")
   @CorrespondsTo("concat")
   @varargs def route(alternatives: Route*): Route = RouteAdapter {
     import akka.http.scaladsl.server.Directives._
 
+    if (alternatives.isEmpty)
+      throw new IllegalArgumentException("Chaining empty list of routes is illegal.")
+
     alternatives.map(_.delegate).reduce(_ ~ _)
   }
 
+  /**
+   * Used to chain multiple alternate routes using comma,
+   * rather than having to explicitly call route1.orElse(route2).orElse(route3).
+   */
+  @CorrespondsTo("concat")
```

Comment:
Do we need the `@CorrespondsTo` when they have the same name?

Reply:
Nope, it's only when names are different